test/commands_test.go

```diff
@@ -0,0 +1,68 @@
+// +build content
+
+package test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"testing"
+	"text/template"
+
+	"github.com/ghodss/yaml"
+)
+
+type prow struct {
+	Presubmits struct {
+		OpenshiftOpenshiftAzure []struct {
+			RerunCommand string `json:"rerun_command"`
+		} `json:"openshift/openshift-azure"`
+	} `json:"presubmits"`
+}
+
+func get(url string) ([]byte, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code %d", resp.StatusCode)
+	}
+
+	return ioutil.ReadAll(resp.Body)
+}
+
+func TestGenerateCommands(t *testing.T) {
+	b, err := get("https://raw.githubusercontent.com/openshift/release/master/ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml")
+	if err != nil {
+		t.Error(err)
+	}
+
+	var prow prow
+	var commands []string
+	err = yaml.Unmarshal(b, &prow)
+	if err != nil {
+		t.Error(err)
+	}
+
+	for _, c := range prow.Presubmits.OpenshiftOpenshiftAzure {
+		commands = append(commands, c.RerunCommand)
+	}
+
+	f, err := os.OpenFile("../docs/commands.md",
```

Comment:
use `os.Create()`?

Reply:
Oh, yeah, this is it... opened with APPEND mode below is the source of the bug.